cdap-api/src/main/java/co/cask/cdap/api/schedule/Schedule.java

```diff
@@ -74,30 +88,17 @@ public boolean equals(final Object o) {
       return false;
     }
 
-    Schedule schedule = (Schedule) o;
-
-    if (cronEntry != null
-          ? !cronEntry.equals(schedule.cronEntry)
-          : schedule.cronEntry != null) {
-      return false;
-    }
-    if (description != null ? !description.equals(schedule.description) :
-         schedule.description != null) {
-      return false;
-    }
-    if (name != null ? !name.equals(schedule.name) : schedule.name != null) {
-      return false;
-    }
+    Schedule that = (Schedule) o;
 
-    return true;
+    return Objects.equals(name, that.name) &&
+      Objects.equals(description, that.description) &&
+      Objects.equals(cronEntry, that.cronEntry) &&
+      Objects.equals(runConstraints, that.runConstraints);
```

Comment:
would a schedule with RunConstraints.NONE be equal to a schedule without RunConstraints? Not sure where this is used, but above you are treating them the same for backwards-compatibility.